platforms/core-runtime/launcher/src/main/java/org/gradle/launcher/cli/InProcessUserInputHandlingExecutor.java

```diff
@@ -45,19 +45,32 @@ public InProcessUserInputHandlingExecutor(GlobalUserInputReceiver userInputRecei
 
     @Override
     public BuildActionResult execute(BuildAction action, BuildActionParameters actionParameters, BuildRequestContext buildRequestContext) {
-        userInputReceiver.dispatchTo(() -> {
-            // Read a single line of text from stdin and forward to the UserInputReader
-            String line;
-            try {
-                line = new BufferedReader(new InputStreamReader(System.in)).readLine();
-            } catch (IOException e) {
-                throw new RuntimeException(e);
-            }
-            if (line != null) {
-                userInputReader.putInput(new UserInputReader.TextResponse(line));
-            } else {
-                userInputReader.putInput(UserInputReader.END_OF_INPUT);
-            }
+        userInputReceiver.dispatchTo(normalizer -> {
+            Thread thread = new Thread(() -> {
```

Comment:
💭 This took a couple of reads to understand.  I think a named class would be better here.

Reply:
Done